compiler/rustc_middle/src/mir/interpret/error.rs

```diff
@@ -148,8 +148,6 @@ pub enum InvalidProgramInfo<'tcx> {
     /// (which unfortunately typeck does not reject).
     /// Not using `FnAbiError` as that contains a nested `LayoutError`.
     FnAbiAdjustForForeignAbi(call::AdjustForForeignAbiError),
-    /// An invalid transmute happened.
-    TransmuteSizeDiff(Ty<'tcx>, Ty<'tcx>),
```

Comment:
That's great. :D So does this also fix https://github.com/rust-lang/rust/issues/79047 ?

What about `SizeOfUnsizedType`, which seems similar to me?

Reply:
> That's great. :D So does this also fix #79047 ?

yes, the corresponding test has been updated

Reply:
We might want to keep the issue open for `SizeOfUnsizedType` though.

Reply:
> What about SizeOfUnsizedType, which seems similar to me?

I have opened https://github.com/rust-lang/rust/issues/97477 for this.